### PR TITLE
DAC-350 Time-run beneath the Labels in the timeline

### PR DESCRIPTION
**Pull request by amnambiar**
https://input-output.atlassian.net/browse/DAC-350

### CODE REVIEW

File: react-web/src/compositions/Timeline/Timeline.tsx [hunk 1]
```diff
@@ -1,19 +1,26 @@
-import React from "react";
+import React, { useEffect } from "react";
```
Comment (bogdan-manole) [inline react-web/src/compositions/Timeline/Timeline.tsx:1]: where `useEffect` is used?
don't you use any linter (eg elsint) that warns you about unnecessary code blocks?

File: react-web/src/hooks/useLogs.ts [hunk 2]
```diff
@@ -13,41 +14,119 @@ export const useLogs = (
...
+        if (testEnded) { 
+            dispatch(setEnded(ended+1)); 
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [testEnded])
+
+
+    useEffect(() => {
+        setEnabled(!fetchingLogs && !(ended > 1) && !!uuid)
+    }, [fetchingLogs, ended, uuid])
```
Comment (bogdan-manole) [inline react-web/src/hooks/useLogs.ts:35]: unnecessarily and possibly harmful way of using `useEffect` as a state changer.
https://beta.reactjs.org/learn/you-might-not-need-an-effect

```suggestion
   const enabled = !fetchingLogs && !(ended > 1) && !!uuid
```

File: react-web/src/hooks/useLogs.ts [hunk 3]
```diff
@@ -13,41 +14,119 @@ export const useLogs = (
     testEnded: boolean,
     handleErrorScenario: () => void
 ) => {
+    const dispatch = useDispatch()
     const [logInfo, setLogInfo] = useState<Log[]>([])
     const [fetchingLogs, setFetchingLogs] = useState(false);
     const [refetchLogsOffset] = useState(1);
+    const [enabled, setEnabled] = useState(false)
```
Comment (bogdan-manole) [inline react-web/src/hooks/useLogs.ts:21]: see change request from below (line 33) for more details
```suggestion
```

File: react-web/src/hooks/useLogs.ts [hunk 4]
```diff
@@ -13,41 +14,119 @@ export const useLogs = (
...
+
+    const computeRunState = (state: string) => {
+        if (state.indexOf('build-flake') !== -1) { 
+            return 'building'
+        } else if (state.indexOf('generate-flake') !== -1) {
+            return 'preparing'
+        } else if (state.indexOf('run-certify') !== -1) {
+            return 'certifying'
+        } else {
+            return '' ;
```
Comment (bogdan-manole) [inline react-web/src/hooks/useLogs.ts:58]: mixing rule of using `;` with other places of not using; I strongly recommend starting the use of `eslint` and enforcing some consistent set of rules

File: react-web/src/pages/certification/Certification.tsx [hunk 5]
```diff
@@ -315,11 +317,10 @@ const Certification = () => {
               hasFailedTasks={isAnyTaskFailure(resultData)}
             />
           </div>
-          {runState ? (
-            <>
-              <InformationTable logs={logInfo} />
-            </>
-          ) : null}
+          
+          {/* {runState ? (<><InformationTable logs={logInfo} /></>) : null} */}
```
Comment (bogdan-manole) [inline react-web/src/pages/certification/Certification.tsx:321]: do we need this commented line?
```suggestion
```

File: react-web/src/utils/utils.ts [hunk 6]
```diff
@@ -33,4 +33,26 @@ export const formatToTitleCase = (value: string) => {
     const finalResult = result.charAt(0).toUpperCase() + result.slice(1);
     return finalResult.trim();
   }
+}
+
+export const formatTimeToReadable = (duration: number) => {
+    var milliseconds = Math.floor(duration % 1000), //Math.floor((duration % 1000) / 100),
```
Comment (bogdan-manole) [inline react-web/src/utils/utils.ts:39]: 1. same for this comment. do we plan to change this computation?
2. why var?
```suggestion
    const milliseconds = Math.floor(duration % 1000),
```

**Review by bogdan-manole — CHANGES_REQUESTED** [review]

File: react-web/src/compositions/Timeline/Timeline.tsx (hunk 1 above)
Comment (amnambiar) [reply react-web/src/compositions/Timeline/Timeline.tsx]: Removed,

File: react-web/src/hooks/useLogs.ts (hunk 4 above)
Comment (amnambiar) [reply react-web/src/hooks/useLogs.ts]: Corrected this instance. Will ensure in future, as well as correct the other places with time.

File: react-web/src/pages/certification/Certification.tsx (hunk 5 above)
Comment (amnambiar) [reply react-web/src/pages/certification/Certification.tsx]: Replaced with another comment

File: react-web/src/utils/utils.ts (hunk 6 above)
Comment (amnambiar) [reply react-web/src/utils/utils.ts]: Missed to update the code, Corrected.

File: react-web/src/hooks/useLogs.ts (hunk 2 above)
Comment (amnambiar) [reply react-web/src/hooks/useLogs.ts:35]: I do understand this suggestion, and it was in a const variable before moving it into state var in this PR. I did this, as part of performance optimization to minimize recalculations. I will recheck this and get back soon.

Comment (amnambiar) [reply react-web/src/hooks/useLogs.ts]: @bogdan-manole I've done this change, pls re-review

File: react-web/src/hooks/useLogs.ts (hunk 3 above)
Comment (amnambiar) [reply react-web/src/hooks/useLogs.ts]: Done

**Review by bogdan-manole — APPROVED** [review]

**Pull request merged**